src/mqt/predictor/rl/predictorenv.py

```diff
@@ -50,6 +50,11 @@ def __init__(
         self.actions_final_optimization_indices = []
         self.used_actions: list[str] = []
         self.device = get_device_by_name(device_name)
+        # check whether bi-directional
+        for a, b in self.device.coupling_map:
+            if [b, a] not in self.device.coupling_map:
+                msg = f"The connectivity of device '{device_name}' is not bi-directional which is not supported by the MQT Predictor."
+                raise ValueError(msg)
```

Comment:
I think I would prefer to not error out here, but simply warn the user. It might as well be that everything works just fine. We just do not guarantee it and the user is on their own.
Simply warning here and then breaking out of the loop and continuing with the execution seems a little more convenient here.

These lines should also be covered by a test.

Reply:
I converted it into a warning and added a respective check.